[17기 박재현] 장바구니 로직 변경

namujinju opened the pull request:
>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
- 주문 관련  오류 수정
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- order, orderproduct 예외 처리
- 주문 로직이 제대로 작동하지 않은 점 수정

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [x] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [x] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [x] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [x] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

namujinju committed:
Refactor: ProductCartView

jamessoun93 commented on product/views.py:
```diff
@@ -123,10 +123,9 @@ def post(self, request):
-                for order_product in order_products:
-                    order_product.quantity+=quantity
-                    order_product.save()
+                order_product = OrderProduct.objects.get(order=order, product_option=product_option)
+                order_product.quantity+=quantity
+                order_product.save()
```
get() 메소드 쓰시면 꼭 exception 처리 해주셔야합니다. 어떤 exception 발생할 수 있는지 확인 후 처리해주세요.

namujinju replied:
수정 완료했습니다.

namujinju committed:
Modify: 예외처리 추가

namujinju committed:
Add: 누락된 order 앱 url 추가

jamessoun93 commented:
conflict 해결해주세요.

jamessoun93 commented on order/views.py:
```diff
@@ -98,4 +98,7 @@ def post(self, request):
\ No newline at end of file
+            return JsonResponse({'message':'KEY_ERROR'}, status=400)
+        
+        except OrderProduct.DoesNotExist:
+            return JsonResponse({'message':'INVALID_ORDER'}, status=400)
```
get 메소드가 여러개의 객체를 반환할 시 raise 되는 exception 도 있습니다. 추가해주세요!

namujinju replied:
추가 완료했습니다! conflict도 해결했습니다.

namujinju replied:
주문이 제대로 동작하지 않아서 주문 로직도 변경했습니다.

namujinju committed:
Modify: 예외 처리
- Order와 OrderProduct가 존재하지 않거나 여러 값을 불러오는 경우 예외
처리
- 없는 class를 import 하는 오류 수정

namujinju committed:
Add: migrations file

namujinju committed:
Modify: Order 모델 다시 추가

namujinju committed:
Modify: 주문 관련 로직 변경
- 주문 과정에서 오류가 있었던 부분 수정

namujinju committed:
Fix: 테스트용 코드 삭제

namujinju committed:
Fix: 불필요한 공백 제거

namujinju committed:
Modify: 테스트용 코드 제거

namujinju committed:
Modify: 테스트용 코드 제거

jamessoun93 commented:
conflict 해결해주세요.

jamessoun93 commented on product/views.py:
```diff
@@ -98,29 +98,31 @@ def get(self, request):
-
+            # user = request.user
+            user = User.objects.get(id=1)
+            data       = json.loads(request.body)
```
마찬가지입니다.

jamessoun93 commented on order/views.py:
```diff
@@ -15,6 +15,7 @@ class OrderProductView(View):
     def get(self, request):
         try:
             user = request.user
+            user = User.objects.get(id=1)
```
위에서 이미 유저객체 가지고 오는데 유저 하드코딩해서 받아오시는 방식으로 수정하신 이유가 있나요?

namujinju replied:
테스트를 위해 임시로 설정했습니다. 45분 전에 최종적으로 삭제해서 commit해서 push 했습니다. 혹시 적용이 안된 건가요?

jamessoun93 commented on order/views.py:
```diff
@@ -65,11 +66,11 @@ def get(self, request):
     def post(self, request):
         try:
             user = request.user
+            user = User.objects.get(id=1)
```
여기도 있네요.

namujinju committed:
Merge branch 'main' into feature/productcart
